kernel/kernel-api/src/main/java/io/delta/kernel/internal/types/DataTypeJsonSerDe.java

```diff
@@ -341,6 +372,36 @@ private static String getStringField(JsonNode rootNode, String fieldName) {
     return node.textValue(); // double check this only works for string values! and isTextual()!
   }
 
+  private static void assertValidTypeForCollations(
+      String fieldPath, String fieldType, Map<String, String> collationMap) {
+    if (collationMap.containsKey(fieldPath) && !fieldType.equals("string")) {
+      throw new IllegalArgumentException(String.format("Invalid data type for collations: \"%s\"", fieldType));
+    }
+  }
+
+  /**
+   * Returns a map of field path to collation name.
+   */
+  private static HashMap<String, String> getCollationsMap(JsonNode fieldMetadata) {
+    if (fieldMetadata == null || !fieldMetadata.has(DataType.COLLATIONS_METADATA_KEY)) {
+      return new HashMap<>();
+    }
+    HashMap<String, String> collationsMap = new HashMap<>();
+    FieldMetadata collationFieldMetadata =
+        parseFieldMetadata(fieldMetadata.get(DataType.COLLATIONS_METADATA_KEY));
+    for (Map.Entry<String, Object> collationField :
+        collationFieldMetadata.getEntries().entrySet()) {
+      String fieldPath = collationField.getKey();
+      Object collationName = collationField.getValue();
+      if (!(collationName instanceof String)) {
+        throw new IllegalArgumentException(
+            String.format("Invalid collation name: %s.", collationName));
+      }
+      collationsMap.put(fieldPath, (String) collationName);
+    }
+    return collationsMap;
+  }
```

Comment:
I don't see a need for this. Basically instead of creating the Map, why not just use the `fieldMetadata`? And the `fieldMetadata` has a `getString` which already has type checks. 

This will improve the code in this file as well. I think we do similarly in ColumnMapping.java for nested field ids.

Reply:
you're right, thanks!